### DISCUSSION TITLE
can not delete element from xml config

### DISCUSSION THREAD

**Discussion by Mentopolis** [wixtoolset/issues#6383] [Questions]
Hello,

using Wix Toolset i cant delete an xml element from an xml file. All details are posted here. There are some tests posted but we already work on this multiple days and find no solutions.

How to delete the "DrawListAddin" from this list with WIX XmlConfig ?

      <?xml version="1.0" encoding="utf-8"?>
      <ArrayOfString xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema- instance">
        <string>ExplorerAddin</string>
        <string>DrawListAddin</string>
        <string>MyDataAddin</string>
      </ArrayOfString>

Result should look like this

      <?xml version="1.0" encoding="utf-8"?>
      <ArrayOfString xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema- instance">
        <string>ExplorerAddin</string>
        <string>MyDataAddin</string>
      </ArrayOfString>

i try multiple settings but nothing work .. the last version looks like this. Searching for the correct item and than remove it. The XML file gets modified but no changes.

    <util:XmlConfig Id='XmlSettingsRemove' 
	    On="install"
		File='[E3D31_INSTALL_DIR]DesignAddins.xml' 
	    Node='element'
		Action='delete' 						
		VerifyPath="//*[\[]text() [\[]contains(.,'DrawListAddin')[\]][\]]" 
		ElementPath="//ArrayOfString" 
		Sequence='1' 
	 />

this code bring an error message that the node could not be found but the "/.." at the end explicit selects the parent correct

    <util:XmlConfig Id='XmlSettingsRemove' 
      On="install"
      File='[E3D31_INSTALL_DIR]DesignAddins.xml' 
      Node='element'
      Action='delete' 						
      VerifyPath="/ArrayOfString/string[\[]text()[\[]contains(.,'DrawListAddin')[\]][\]]" 
      ElementPath="/ArrayOfString/string[\[]text()[\[]contains(.,'DrawListAddin')[\]][\]]/.." 
      Sequence='1' 
    	/>

also here : https://stackoverflow.com/questions/66479166/wix-installer-remove-entry-from-xmlconfig-element

regards
  Markus

**Comment by barnson** [comment]
WiX doesn't use issues for support. We moved this issue to the [experimental GitHub discussions area][gh].

[gh]: https://github.com/wixtoolset/issues/discussions?discussions_q=category%3AQuestions
